### PR TITLE
Parse ADT example 

### PR DESCRIPTION
**Pull request by reshmakh**
Proposed for #4489

### CODE REVIEW

File: examples/medplum-demo-bots/src/adt-transfer-listener/adt-transfer-bot.test.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import { Hl7Message, indexSearchParameterBundle, indexStructureDefinitionBundle } from '@medplum/core';
+import { Bot, Bundle, Reference, SearchParameter } from '@medplum/fhirtypes';
+import { MockClient } from '@medplum/mock';
+import { expect, test } from 'vitest';
+import { handler } from './adt-transfer-bot';
+import { SEARCH_PARAMETER_BUNDLE_FILES, readJson } from '@medplum/definitions';
+
+//To run these tests from the command line
+//npm t src/adt-transfer-listener/adt-transfer-bot.test.ts
```
Comment (codyebberson) [inline examples/medplum-demo-bots/src/adt-transfer-listener/adt-transfer-bot.test.ts:9]:
```suggestion
// To run these tests from the command line
// npm t src/adt-transfer-listener/adt-transfer-bot.test.ts
```

File: examples/medplum-demo-bots/src/adt-transfer-listener/adt-transfer-bot.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  // Log Message Type
+  const messageType = input.getSegment('MSH')?.getField(9)?.getComponent(1) as string;
+  const messageSubtype = input.getSegment('MSH')?.getField(9)?.getComponent(2) as string;
+
+  // If this is anything but ADT
+  if (messageType !== 'ADT') {
+    return input.buildAck();
+  }
+
+  //Only supported message types should be processed
```
Comment (codyebberson) [inline examples/medplum-demo-bots/src/adt-transfer-listener/adt-transfer-bot.ts:17]:
```suggestion
  // Only supported message types should be processed
```

File: examples/medplum-demo-bots/src/adt-transfer-listener/adt-transfer-bot.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import { BotEvent, Hl7Message, MedplumClient } from '@medplum/core';
+import { Encounter, Patient } from '@medplum/fhirtypes';
+
+export async function handler(medplum: MedplumClient, event: BotEvent): Promise<Hl7Message> {
```
Comment (codyebberson) [inline examples/medplum-demo-bots/src/adt-transfer-listener/adt-transfer-bot.ts:4]:
```suggestion
export async function handler(medplum: MedplumClient, event: BotEvent<Hl7Message>): Promise<Hl7Message> {
```

File: examples/medplum-demo-bots/src/adt-transfer-listener/adt-transfer-bot.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  const input = event.input as Hl7Message;
```
Comment (codyebberson) [inline examples/medplum-demo-bots/src/adt-transfer-listener/adt-transfer-bot.ts:5]:
```suggestion
  const input = event.input;
```

**Review by codyebberson — APPROVED** [review]
Nice, overall looks good.  A bunch of minor stuff, but feel free to ignore.

Some of these would make great HL7 helpers in `@medplum/core`, such as `PID-to-Patient`.  For future consideration

**Comment by reshmakh** [comment]
Created helper task for PID to Patient for review/feedback https://github.com/medplum/medplum/issues/4599

**Pull request merged**